src/vue/forms/IssuanceForm.vue

```diff
@@ -0,0 +1,272 @@
+          label: `${asset.details.name} (${asset.code})`
+        }))
+    },
+    availableAmount () {
+      if (this.ownedAssets.length && this.form.asset) {
+        const asset = this.ownedAssets
+          .find(asset => asset.code === this.form.asset)
+        return {
+          value: asset.availableForIssuance,
+          currency: asset.code
+        }
+      }
+      return { value: 0 }
+    }
+  },
+  async created () {
+    await this.loadOwnedAssets()
+    if (this.ownedAssets.length > 0) {
+      this.form.asset = this.ownedAssets[0].code
+    }
+  },
+  methods: {
+    async submit () {
+      if (!this.isFormValid()) return
+      this.disableForm()
+      try {
+        const receiverBalance =
+          await this.getReceiverBalance(this.form.receiver, this.form.asset)
+
+        if (receiverBalance) {
+          const operation =
+            base.CreateIssuanceRequestBuilder.createIssuanceRequest({
+              asset: this.form.asset,
+              amount: this.form.amount.toString(),
+              receiver: receiverBalance.balanceId,
+              reference: this.form.reference,
+              externalDetails: {}
+            })
+          await Sdk.horizon.transactions.submitOperations(operation)
+          Bus.success('issuance.tokens-issued-msg')
+          this.emitCancel()
+        } else {
+          Bus.error('issuance.balance-required-err')
+        }
+      } catch (e) {
+        console.error(e)
+        ErrorHandler.process(e)
+      }
+      this.enableForm()
+    },
+    async getAccountInfoByEmail (email) {
+      try {
+        const response = await Sdk.api.users.getPage({ email })
+        return response.data.find(info => info.email === email)
+      } catch (e) {
+        console.error(e)
+        ErrorHandler.process(e)
+      }
+    },
+    async getReceiverBalance (receiver, asset) {
+      try {
+        const receiverAccountId = await this.getReceiverAccountId(receiver)
+        const response =
+          await Sdk.horizon.account.getBalances(receiverAccountId)
+        const receiverBalance = response.data
+          .find(balance => balance.asset === asset)
+        return receiverBalance
+      } catch (e) {
+        console.error(e)
+        Bus.error('issuance.wrong-receiver-err')
+      }
+    },
+    async getReceiverAccountId (receiver) {
+      if (receiver.indexOf('@') !== -1) {
+        const receiverAccount = await this.getAccountInfoByEmail(receiver)
+        if (!receiverAccount) throw new Error('Wrong account email')
+        return receiverAccount.id
+      }
+      return receiver
+    },
+    emitCancel () {
+      this.$emit(EVENTS.cancel)
+    }
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+@import './app-form';
+
+.issuance-form__submit-btn {
+  @include button-raised();
+
+  margin-bottom: 2rem;
+  width: 18rem;
+}
+
+.issuance-form__cancel-btn {
+  @include button();
+
+  padding-left: .1rem;
+  padding-right: .1rem;
+  margin-left: auto !important;
```

Comment:
If you use `important`, you should at least add a comment with a description on why have you decided so. This may prevent other programmers from thinking you are bastard

Reply:
Resolved, fixed